services/provider/server/server.go

```diff
@@ -722,3 +734,135 @@ func (s *OCSProviderServer) getOCSSubscriptionChannel(ctx context.Context) (stri
 	}
 	return subscription.Spec.Channel, nil
 }
+
+func (s *OCSProviderServer) OnboardMirroringPeer(ctx context.Context, req *pb.OnboardMirroringPeerRequest) (*pb.OnboardMirroringPeerResponse, error) {
+
+	pubKey, err := s.getOnboardingValidationKey(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get public key to validate onboarding ticket for StorageClusterPeer %q. %v", req.StorageClusterPeerName, err)
+	}
+
+	if err := validateTicket(req.OnboardingTicket, pubKey, services.MirroringPeerRole); err != nil {
+		klog.Errorf("failed to validate onboarding ticket for StorageClusterPeer  %q. %v", req.StorageClusterPeerName, err)
+		return nil, status.Errorf(codes.InvalidArgument, "onboarding ticket is not valid. %v", err)
+	}
+
+	representationID, err := s.storageClusterPeerManager.Create(ctx, req.OnboardingTicket, req.StorageClusterPeerName)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) && err != errTicketAlreadyExistsStorageClusterPeer {
+			return nil, status.Errorf(codes.Internal, "failed to create StorageClusterPeer representation %q. %v", req.StorageClusterPeerName, err)
+		}
+
+		representation, err := s.storageClusterPeerManager.GetByName(ctx, req.StorageClusterPeerName)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Failed to get storageConsumer. %v", err)
```

Comment:
failed to retrieve the storage consumer. Did you mean storageClusterPeer instead?